### PR TITLE
simulator host route setup script and docs

### PR DESCRIPTION
**Pull request by adamfeuer**
### Summary

* adds simulator host route setup script
* updates docs to state that this works on Linux and how to use it

### Impact

* none if you don't use the `tools/simhostroute.sh` script
*  `tools/simhostroute.sh` can be used to set up a host route and IP tables rules to allow the nuttx simulator to access the Internet.

### Limitations / TODO

* Having to manually set the capability on the `nuttx` executable is not ideal; a future script could both add the capability and start `nuttx`.

### Detail

* The NuttX simulator is given the IP address `10.0.1.2`; the `nuttx0` tap device on the host is given the IP address `10.0.1.1`. 
* Routes and IP Tables entries are added to do NAT and allow the NuttX simulator to access the Internet.

### Testing

* Manual, and by using tcpecho and tcpblaster

### How To Verify 

* Follow the directions in the `NETWORK-LINUX.txt` file.

### CODE REVIEW

File: tools/simhostroute.sh [hunk 1]
```diff
@@ -0,0 +1,59 @@
+#!/bin/bash
```
Comment (xiaoxiang781216) [inline tools/simhostroute.sh]: Add apache copyright, like tools/zds/zdsar.c

File: tools/simhostroute.sh [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+set -x
+
+# Helper script to set up host route to NuttX simulator
```
Comment (xiaoxiang781216) [inline tools/simhostroute.sh]: Add some description in tools/README.txt

File: tools/simhostroute.sh [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+IP_NUTTX="10.0.1.2"
+
+if [ "$STATUS" == "on" ]; then
+    ip link add $IF_BRIDGE type bridge
+    ifconfig $IF_BRIDGE $IP_HOST
+    ifconfig $IF_BRIDGE up
+    ifconfig -a
+    ip addr add $IP_HOST dev $IF_BRIDGE 
+    ifconfig $IF_BRIDGE netmask $IP_NETMASK
+    ip route delete $IP_NET
```
Comment (xiaoxiang781216) [inline tools/simhostroute.sh]: don't need, done at line 47

File: tools/simhostroute.sh [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+IF_BRIDGE=nuttx0
+IP_NET="10.0.1.0/24"
+IP_NETMASK="255.255.255.0"
+IP_BROADCAST="10.0.0.255"
+IP_HOST="10.0.1.1"
+IP_NUTTX="10.0.1.2"
+
+if [ "$STATUS" == "on" ]; then
+    ip link add $IF_BRIDGE type bridge
```
Comment (xiaoxiang781216) [inline tools/simhostroute.sh]: It seem that we are still using bridge mode? does this script work with CONFIG_SIM_NET_HOST_ROUTE?

File: tools/simhostroute.sh [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    ifconfig $IF_BRIDGE $IP_HOST
```
Comment (xiaoxiang781216) [inline tools/simhostroute.sh]: use two spaces indent

File: tools/simhostroute.sh (hunk 1 above)
Comment (adamfeuer) [reply tools/simhostroute.sh]: @xiaoxiang781216 will fix

File: tools/simhostroute.sh (hunk 2 above)
Comment (adamfeuer) [reply tools/simhostroute.sh]: @xiaoxiang781216 will fix

File: tools/simhostroute.sh (hunk 4 above)
Comment (adamfeuer) [reply tools/simhostroute.sh]: @xiaoxiang781216 It's using host route mode, and should have CONFIG_SIM_NET_HOST_ROUTE set. I'll look at my setup tomorrow and redo the defconfig so it has that.

As for the name `$IF_BRIDGE` it should really be called `$IF_TAP` or `$IF_NUTTX`. I'll fix that too.

File: tools/simhostroute.sh (hunk 5 above)
Comment (adamfeuer) [reply tools/simhostroute.sh]: @xiaoxiang781216 will fix

File: tools/simhostroute.sh (hunk 3 above)
Comment (adamfeuer) [reply tools/simhostroute.sh]: @xiaoxiang781216 I don't understand, will you elaborate?

Comment (adamfeuer) [reply tools/simhostroute.sh]: @xiaoxiang781216 I think I get it. I'll refactor this script to match the netusb.sh script (after fixing that one per your comments).

Comment (xiaoxiang781216) [reply tools/simhostroute.sh]: I mean that "ip route delete $IP_NET" already do in the off path, so we don't need delete again in on path. For example, you don't delete $IP_NUTTX/32 in the on path.

**Comment by adamfeuer** [comment]
@xiaoxiang781216 I made the suggested improvements, rebased, and force-pushed my branch here.

Re: the `defconfig` not having `SIM_NET_HOST_ROUTE` in it, that is the default, so it's left out of the `defconfig` when savedefconfig is done. I tested to make sure the updated script works with the updated defconfig, and it does.

**Pull request merged**